services/motion/builtin/testutils.go

```diff
@@ -0,0 +1,542 @@
+package builtin
+
+import (
+	"context"
+	"math"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.uber.org/multierr"
+	"go.viam.com/test"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/base/kinematicbase"
+	"go.viam.com/rdk/components/base/wheeled"
+	"go.viam.com/rdk/components/encoder"
+	fakeencoder "go.viam.com/rdk/components/encoder/fake"
+	"go.viam.com/rdk/components/motor"
+	fakemotor "go.viam.com/rdk/components/motor/fake"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/components/movementsensor/wheeledodometry"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot/framesystem"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+var (
+	leftMotorName  = "leftmotor"
+	rightMotorName = "rightmotor"
+	baseName       = "test-base"
+	moveSensorName = "test-movement-sensor"
+
+	moveSensorResource        = resource.NewName(movementsensor.API, moveSensorName)
+	baseResource              = resource.NewName(base.API, baseName)
+	movementSensorInBasePoint = r3.Vector{X: -10, Y: 0, Z: 0}
+	updateRate                = 33
+)
+
+func getPointCloudMap(path string) (func() ([]byte, error), error) {
+	const chunkSizeBytes = 1 * 1024 * 1024
+	file, err := os.Open(path) //nolint:gosec
+	if err != nil {
+		return nil, err
+	}
+	chunk := make([]byte, chunkSizeBytes)
+	f := func() ([]byte, error) {
+		bytesRead, err := file.Read(chunk)
+		if err != nil {
+			defer utils.UncheckedErrorFunc(file.Close)
+			return nil, err
+		}
+		return chunk[:bytesRead], err
+	}
+	return f, nil
+}
+
+func createInjectedMovementSensor(name string, gpsPoint *geo.Point) *inject.MovementSensor {
+	injectedMovementSensor := inject.NewMovementSensor(name)
+	injectedMovementSensor.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+		return gpsPoint, 0, nil
+	}
+	injectedMovementSensor.CompassHeadingFunc = func(ctx context.Context, extra map[string]interface{}) (float64, error) {
+		return 0, nil
+	}
+	injectedMovementSensor.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{CompassHeadingSupported: true}, nil
+	}
+
+	return injectedMovementSensor
+}
+
+func createInjectedSlam(name string) *inject.SLAMService {
+	pcdPath := "pointcloud/octagonspace.pcd"
```

Comment:
Only some tests use `octagonspace.pcd`
 others like `TestObstacleReplanningSlam` use `"pointcloud/cardboardOcto.pcd"` or `TestGetTransientDetections` uses `"slam/example_cartographer_outputs/viam-office-02-22-3/pointcloud/pointcloud_4.pcd"`


Is this a non-issue because these tests are moved out of RDK and into motion testing?

Reply:
> Is this a non-issue because these tests are moved out of RDK and into motion testing?

Exactly :)